components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIKeyValidator.java

```diff
@@ -355,18 +359,27 @@ public boolean isAPIResourceValidationEnabled() {
 
     @MethodStats
     public String getResourceAuthenticationScheme(MessageContext synCtx) throws APISecurityException {
-
-        VerbInfoDTO verb = null;
+        String authType = "";
+        List<VerbInfoDTO> verbInfoList;
         TracingSpan span = null;
         try {
             if (Util.tracingEnabled()) {
                 TracingSpan keySpan = (TracingSpan) synCtx.getProperty(APIMgtGatewayConstants.KEY_VALIDATION);
                 TracingTracer tracer = Util.getGlobalTracer();
                 span = Util.startSpan(APIMgtGatewayConstants.FIND_MATCHING_VERB, keySpan, tracer);
             }
-            verb = findMatchingVerb(synCtx);
-            if (verb != null) {
-                synCtx.setProperty(APIConstants.VERB_INFO_DTO, verb);
+            verbInfoList = findMatchingVerb(synCtx);
+            if (verbInfoList != null && verbInfoList.toArray().length > 0) {
+                for (VerbInfoDTO verb : verbInfoList) {
+                    authType = verb.getAuthType();
+                    if (authType == null) {
```

Comment:
Since you have defined the authType as below, we do not want a validate null valisation. 
String authType = "";